README.md

```diff
@@ -1,2 +1,313 @@
+- IAM Role to allow the cluster to access other AWS services
+- Security Group which is used by EKS workers to connect to the cluster and kubelets and pods to receive communication from the cluster control plane (see [terraform-aws-eks-workers](https://github.com/cloudposse/terraform-aws-eks-workers))
+- The module generates `kubeconfig` configuration to connect to the cluster using `kubectl`
+
+## Usage
+
+
+For a complete example, see [examples/complete](examples/complete)
+
+```hcl
+provider "aws" {
+  region = "${var.region}"
+}
+
+locals {
+  # The usage of the specific kubernetes.io/cluster/* resource tags below are required
+  # for EKS and Kubernetes to discover and manage networking resources
+  # https://www.terraform.io/docs/providers/aws/guides/eks-getting-started.html#base-vpc-networking
+  tags = "${merge(var.tags, map("kubernetes.io/cluster/${var.cluster_name}", "shared"))}"
+}
+
+data "aws_availability_zones" "available" {}
+
+module "vpc" {
+  source     = "git::https://github.com/cloudposse/terraform-aws-vpc.git?ref=master"
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  name       = "${var.name}"
+  attributes = "${var.attributes}"
+  tags       = "${local.tags}"
+  cidr_block = "10.0.0.0/16"
+}
+
+module "subnets" {
+  source              = "git::https://github.com/cloudposse/terraform-aws-dynamic-subnets.git?ref=master"
+  availability_zones  = ["${data.aws_availability_zones.available.names}"]
+  namespace           = "${var.namespace}"
+  stage               = "${var.stage}"
+  name                = "${var.name}"
+  attributes          = "${var.attributes}"
+  tags                = "${local.tags}"
+  region              = "${var.region}"
+  vpc_id              = "${module.vpc.vpc_id}"
+  igw_id              = "${module.vpc.igw_id}"
+  cidr_block          = "${module.vpc.vpc_cidr_block}"
+  nat_gateway_enabled = "true"
+}
+
+module "eks_cluster" {
+  source                  = "git::https://github.com/cloudposse/terraform-aws-eks-cluster.git?ref=master"
+  namespace               = "${var.namespace}"
+  stage                   = "${var.stage}"
+  name                    = "${var.name}"
+  attributes              = "${var.attributes}"
+  tags                    = "${var.tags}"
+  vpc_id                  = "${module.vpc.vpc_id}"
+  subnet_ids              = ["${module.subnets.public_subnet_ids}"]
+  allowed_security_groups = ["${var.allowed_security_groups}"]
+  allowed_cidr_blocks     = ["${var.allowed_cidr_blocks}"]
+}
+
+module "eks_workers" {
+  source                             = "git::https://github.com/cloudposse/terraform-aws-eks-workers.git?ref=master"
+  namespace                          = "${var.namespace}"
+  stage                              = "${var.stage}"
+  name                               = "${var.name}"
+  attributes                         = "${var.attributes}"
```

Comment:
These examples aren’t really that helpful. All the variables are defined as other variables that are not defined anywhere :)

Reply:
updated